pkg/yttlibrary/ip.go

```diff
@@ -0,0 +1,142 @@
+type IPAddrValue struct {
+	addr                 net.IP
+	*core.StarlarkStruct // TODO: keep authorship of the interface by delegating instead of embedding
+}
+
+func (av *IPAddrValue) Type() string { return "@ytt:ip.addr" }
+
+func (av *IPAddrValue) AsStarlarkValue() starlark.Value {
+	m := orderedmap.NewMap()
+	m.Set("is_ipv4", starlark.NewBuiltin("ip.addr.is_ipv4", core.ErrWrapper(av.IsIPv4)))
+	m.Set("is_ipv6", starlark.NewBuiltin("ip.addr.is_ipv6", core.ErrWrapper(av.IsIPv6)))
+	m.Set("string", starlark.NewBuiltin("ip.addr.string", core.ErrWrapper(av.string)))
+	av.StarlarkStruct = core.NewStarlarkStruct(m)
+	return av
+}
+
+func (av *IPAddrValue) ConversionHint() string {
+	return "IPAddrValue does not automatically encode (hint: use .string())"
+}
+
+func (av *IPAddrValue) IsIPv4(thread *starlark.Thread, f *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	if args.Len() != 0 {
+		return starlark.None, fmt.Errorf("expected no argument")
+	}
+	isV4 := av.addr != nil && av.addr.To4() != nil
+	return starlark.Bool(isV4), nil
+}
+
+func (av *IPAddrValue) IsIPv6(thread *starlark.Thread, f *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	if args.Len() != 0 {
+		return starlark.None, fmt.Errorf("expected no argument")
+	}
+	isV6 := av.addr != nil && av.addr.To4() == nil
+	return starlark.Bool(isV6), nil
+}
+
+func (av *IPAddrValue) string(thread *starlark.Thread, f *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	if args.Len() != 0 {
+		return starlark.None, fmt.Errorf("expected no argument")
+	}
+	return starlark.String(av.addr.String()), nil
+}
+
+func (m ipModule) ParseCIDR(thread *starlark.Thread, f *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	if args.Len() != 1 {
+		return starlark.None, fmt.Errorf("expected exactly one argument")
+	}
+
+	cidrStr, err := core.NewStarlarkValue(args.Index(0)).AsString()
+	if err != nil {
+		return starlark.None, err
+	}
+
+	parsedIP, parsedNet, err := net.ParseCIDR(cidrStr)
+	if err != nil {
+		return starlark.None, err
+	}
+
+	return starlark.Tuple{(&IPAddrValue{parsedIP, nil}).AsStarlarkValue(), (&IPNetValue{parsedNet, nil}).AsStarlarkValue()}, nil
+}
+
+type IPNetValue struct {
+	net                  *net.IPNet
+	*core.StarlarkStruct // TODO: keep authorship of the interface by delegating instead of embedding
+}
+
+func (inv *IPNetValue) Type() string { return "@ytt:ip.net" }
+
+func (inv *IPNetValue) AsStarlarkValue() starlark.Value {
+	m := orderedmap.NewMap()
+	m.Set("addr", starlark.NewBuiltin("ip.net.addr", core.ErrWrapper(inv.Addr)))
+	m.Set("string", starlark.NewBuiltin("ip.net.string", core.ErrWrapper(inv.string)))
+	inv.StarlarkStruct = core.NewStarlarkStruct(m)
+	return inv
+}
+
+func (inv *IPNetValue) ConversionHint() string {
+	return "IPNetValue does not automatically encode (hint: use .string())"
```

Comment:
```suggestion
	return inv.Type() + " does not automatically encode (hint: use .string())"
```

Let's refer to the type using the same string that the Starlark `type()` would return.

Reply:
fixed